modules/core/src/test/java/org/apache/ignite/internal/metric/MetricsConfigurationTest.java

```diff
@@ -0,0 +1,174 @@
+
+            assertThrowsWithCause(
+                () -> bean.configureHitRateMetric("io.dataregion.default.AllocationRate", -1),
+                IllegalArgumentException.class);
+
+            bean.configureHitRateMetric("io.dataregion.default.AllocationRate", 5000);
+
+            HitRateMetric allocationRate = g.context().metric().registry(metricName("io.dataregion.default"))
+                .findMetric("AllocationRate");
+
+            assertEquals(5000, allocationRate.rateTimeInterval());
+        }
+    }
+
+    /** Tests configuration of {@link HistogramMetric}. */
+    @Test
+    public void testHistogramConfiguration() throws Exception {
+        try (IgniteEx g = startGrid(0)) {
+            IgniteMXBean bean = (IgniteMXBean)g;
+
+            long[] bounds = new long[] {50, 100};
+
+            //Empty name.
+            assertThrowsWithCause(() -> bean.configureHistogramMetric(null, bounds), NullPointerException.class);
+
+            //Wrong bounds value.
+            assertThrowsWithCause(
+                () -> bean.configureHistogramMetric(metricName(TX_METRICS, METRIC_SYSTEM_TIME_HISTOGRAM), null),
+                NullPointerException.class);
+
+            assertThrowsWithCause(
+                () -> bean.configureHistogramMetric(metricName(TX_METRICS, METRIC_SYSTEM_TIME_HISTOGRAM), new long[0]),
+                IllegalArgumentException.class);
+
+            bean.configureHistogramMetric(metricName(TX_METRICS, METRIC_SYSTEM_TIME_HISTOGRAM), bounds);
+
+            HistogramMetric systemTime = g.context().metric().registry(TX_METRICS)
+                .findMetric(METRIC_SYSTEM_TIME_HISTOGRAM);
+
+            assertArrayEquals(bounds, systemTime.bounds());
+        }
+    }
+
+    /** Tests metric configuration applied on all nodes. */
+    @Test
+    public void testConfigurationSeveralNodes() throws Exception {
+        try (IgniteEx g0 = startGrid(0); IgniteEx g1 = startGrid(1)) {
+            long[] bounds = new long[] {50, 100};
+
+            assertNotEquals(bounds.length, g0.context().metric().registry(TX_METRICS)
+                .<HistogramMetric>findMetric(METRIC_SYSTEM_TIME_HISTOGRAM).bounds().length);
+
+            assertNotEquals(bounds.length, g1.context().metric().registry(TX_METRICS)
+                .<HistogramMetric>findMetric(METRIC_SYSTEM_TIME_HISTOGRAM).bounds().length);
+
+            ((IgniteMXBean)g0).configureHistogramMetric(metricName(TX_METRICS, METRIC_SYSTEM_TIME_HISTOGRAM), bounds);
+
+            assertArrayEquals(bounds, g0.context().metric().registry(TX_METRICS)
+                .<HistogramMetric>findMetric(METRIC_SYSTEM_TIME_HISTOGRAM).bounds());
+
+            assertArrayEquals(bounds, g1.context().metric().registry(TX_METRICS)
+                .<HistogramMetric>findMetric(METRIC_SYSTEM_TIME_HISTOGRAM).bounds());
+        }
+    }
+
+    /** Tests metric configuration applied on all nodes. */
+    @Test
+    public void testNodeRestart() throws Exception {
```

Comment:
I suggest adding a client node to the test (because some histograms like the pme duration presents at clients).
Also, I think we should throw a user-friendly message when trying to configure metrics on a disconnected client.